pinot-common/src/main/java/org/apache/pinot/common/utils/regex/JavaUtilPattern.java

```diff
@@ -20,12 +20,17 @@
 
 import java.util.regex.Pattern;
 
-
 public class JavaUtilPattern implements org.apache.pinot.common.utils.regex.Pattern {
   Pattern _pattern;
 
   public JavaUtilPattern(String regex) {
-    _pattern = Pattern.compile(regex, Pattern.UNICODE_CASE | Pattern.CASE_INSENSITIVE);
```

Comment:
Are we doing case insensitive match now without `FST`? If so, we are already having inconsistent behavior right?